### ISSUE TITLE
Missing WCS keywords in source maps HDUs from gtsrcmaps

### ISSUE THREAD

**Issue by tibaldo** [fermi-lat/Likelihood#29]
The HDUs of the model source maps in output from gtsrcmaps do not have WCS keywords in the header. It seems that the Science Tools read them just fine, which probably is possible because all the WCS information is stored in the header of the primary HDU (the count map) that has the same geometry. However, this makes these HDUs not properly readable by standard astronomy tools such as ds9. I am sure that at least up to the Science Tools 10- series the WCS info was available since I used to read them through ds9 quite often. I’d be curious to know: why where the WCS keywords removed, and could it make sense to restore them? 
This issue was previously posted to the Slack channel and LAT collaboration helpsoftlist. @jballet confirmed the issue and concurred with me that the missing keywords should be restored.

**Comment by jasercion** [comment]
Hi @tibaldo, I'm working on tracking down where and why this change occurred.  If you have a small enough test analysis would you mind testing this with the ScienceTools v11 series that was released before the switch to conda and seeing if the keywords are properly written?  The old downloads are still available on this page:

https://fermi.gsfc.nasa.gov/ssc/data/analysis/software/v11r5p3.html

**Comment by jballet** [comment]
Excellent question.

Indeed those keywords do not exist in source maps created with ST 11-07-00 either.

I had to go back to 2016 to find source maps with the WCS keywords.

Unfortunately the Science Tools do not write in the file their version number, so I cannot be sure, but I think that at the time I was using ScienceTools-HEAD-1066, if it can be of any help.



________________________________
De : jasercion [notifications@github.com]
Envoyé : jeudi 17 janvier 2019 20:23
À : fermi-lat/Likelihood
Cc : Ballet Jean; Mention
Objet : Re: [fermi-lat/Likelihood] Missing WCS keywords in source maps HDUs from gtsrcmaps (#29)


Hi @tibaldo<https://github.com/tibaldo>, I'm working on tracking down where and why this change occurred. If you have a small enough test analysis would you mind testing this with the ScienceTools v11 series that was released before the switch to conda and seeing if the keywords are properly written? The old downloads are still available on this page:

https://fermi.gsfc.nasa.gov/ssc/data/analysis/software/v11r5p3.html

**Comment by Areustle** [comment]
When we moved to supporting healpix maps we altered the class holding maps data to no longer redundantly contain the projection information. As such we do not copy that data into the sub headers. We think implementing it would take a little bit of care to ensure we didn't break something else. Is this a high priority for you @tibaldo  @jballet or do you prefer us to work on other important tasks.

**Comment by tibaldo** [comment]
Hi @Areustle , thank you for asking. I cannot answer about priorities. However, I would certainly recommend to put back in the headers all the information needed to make the HDUs readable by standard astronomy software/tools. Perhaps these keywords are not necessary within the ST as you say, but they are not redundant! Their absence is a real issue in terms of interoperability and user needs. Just to give one example, at present standard visualization tools like ds9 cannot be used to inspect the source maps and track down issues in the analysis.

**Comment by jballet** [comment]
I fully agree with Luigi, and I don't see what you could break by adding back those keywords.

**Comment by jasercion** [comment]
Hi @tibaldo and @jballet - when you have a moment please install the dev version of the Fermitools 
`conda install -c conda-forge -c fermi -c fermi/label/dev fermitools=2.5.88` 
and verify the WCS keywords are now being written out properly to the generated source maps.  Once I hear from you, so long as everything looks ok, I'll merge the change into master for the next release.

**Comment by tibaldo** [comment]
Hi @jasercion, thank you very much for taking care of this! I run a very simple test and everything seems to be working fine. Below a ds9 screenshot with on the left the srcmap entry corresponding to the input map on the right. The crosshair is locked to the same WCS between the two panels, which shows that everything is self-consistent.

<img width="766" height="481" alt="Image" src="https://github.com/user-attachments/assets/d4d4348e-26a9-4019-a980-7ee34a044766" />

**Comment by jasercion** [comment]
Great, thanks for verifying this!  I'll get this closed out and included in the next release.
